### PR TITLE
Add 'CsWinRTEnableIDynamicInterfaceCastableSupport' feature switch

### PR DESCRIPTION
**Pull request by Sergio0694**
This PR adds the 'CsWinRTEnableIDynamicInterfaceCastableSupport' feature switch to allow trimming of all IDIC-related code. This would also include fixing the size regression introduced by #1566 (which can't be fixed without making the code non trim-safe).

> [!NOTE]
> Opening as draft until I get a sizoscope diff.

### CODE REVIEW

**Comment by dongle-the-gadget** [comment]
This would break `As<T>` right?

**Comment by Sergio0694** [comment]
Only if it gets to the fallback IDIC cast from `IInspectable` to the interface type, otherwise it'll still work.

**Comment by manodasanW** [comment]
Note that this is just a feature switch for size improvements if you do not need the IDIC functionality.  If you do need it, you would have the feature switch on which is also the default.

**Comment by dongle-the-gadget** [comment]
> Only if it gets to the fallback IDIC cast from IInspectable to the interface type, otherwise it'll still work

The only case this would be true is if the interface is using built-in COM Interop, which already doesn't support NativeAOT anyway.

**Comment by Sergio0694** [comment]
Saves **52 KB** in the minimal sample, plus more incremental savings as you have more enumerable types 🚀

image

File: src/WinRT.Runtime/Projections/INotifyDataErrorInfo.net5.cs [hunk 1]
```diff
@@ -33,6 +33,13 @@ public static unsafe bool get_HasErrors(IObjectReference obj)
 
         public static unsafe global::System.Collections.IEnumerable GetErrors(IObjectReference obj, string propertyName)
         {
+            if (!FeatureSwitches.IsDynamicObjectsSupportEnabled)
```
Comment (manodasanW) [inline src/WinRT.Runtime/Projections/INotifyDataErrorInfo.net5.cs:36]: EnableIDynamicInterfaceCastableSupport?

Comment (Sergio0694) [reply src/WinRT.Runtime/Projections/INotifyDataErrorInfo.net5.cs]: Oops, fixed in d3ffb5c9620ca9727ca6ed47987e96dc1c233288, thank you! Also renamed the one for dynamic objects to make it consistent with the other names, that's why I tripped up and had accidentally used it here 😅

**Review by manodasanW — APPROVED** [review]

**Pull request merged**